src/Operators/finitedifference.jl

```diff
@@ -285,14 +307,67 @@ struct CompositeStencilStyle <: AbstractStencilStyle end
 
 
 """
-    stencil_interior_width(op, args...)
+    return_eltype(::Op, fields...)
+
+Defines the element type of the result of operator `Op`
+"""
+function return_eltype end
+
+"""
+    return_space(::Op, spaces...)
+
+Defines the space the operator `Op` returns values on.
+"""
+function return_space end
 
-The width of the interior stencil for the operator with the given arguments.
-Returns a tuple of 2-tuples: each 2-tuple should be the lower and upper bounds
-of the index offsets of the stencil for each argument in the stencil.
+"""
+    stencil_interior_width(::Op, args...)
+
+Defines the width of the interior stencil for the operator `Op` with the given
+arguments. Returns a tuple of 2-tuples: each 2-tuple should be the lower and
+upper bounds of the index offsets of the stencil for each argument in the
+stencil.
+
+## Example
+```
+stencil(::Op, arg1, arg2) = ((-half, 1+half), (0,0))
+```
+implies that at index `i`, the stencil accesses `arg1` at `i-half`, `i+half` and
+`i+1+half`, and `arg2` at index `i`.
 """
 function stencil_interior_width end
 
+"""
+    stencil_interior(::Op, loc, idx, args...)
+
+Defines the stencil of the operator `Op` in the interior of the domain at `idx`,
+`args` are the input arguments.
```

Comment:
Can we please clarify this sentence? Is the last bit extra information? As in:

```suggestion
Defines the stencil of the operator `Op` in the interior of the domain at `idx`;
`args` are the input arguments.
```
?

And can we describe `loc` as well?